### PR TITLE
Add dependabot

### PR DESCRIPTION
**Pull request by TheLimeGlass**
### Description
Adds GitHub's Dependabot support.

Automatically keeps the gradle.build file dependencies up to date.

If a newer version of a dependency causes our tests to fail, dependabot will open an issue/pull and we'll have to fix it before it can merge.

Been using Dependabot for years now, it's very helpful. You can rebase a commit with `@dependabot rebase` in the issue when changes have been made to fix it, and it'll build the new dependency with the updated master branch.

---
**Related Issues:** (https://github.com/SkriptLang/Skript/issues/3957)

TP did a test on their master, you can view what the pull requests will look like here https://github.com/TPGamesNL/Skript/pulls

### CODE REVIEW

File: .github/dependabot.yml
Comment (skmedix) [inline .github/dependabot.yml:7]: Wouldn't hurt to check on github actions too ✌
```suggestion
  open-pull-requests-limit: 10
- package-ecosystem: github-actions
  directory: "/"
  schedule:
    interval: daily
  open-pull-requests-limit: 10
```

Comment (TheLimeGlass) [reply .github/dependabot.yml:7]: What does this do?

Comment (skmedix) [reply .github/dependabot.yml:7]: It's checking on action versions used in workflows, just like you bumped actions in #4845.

Docs: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot

File: .github/workflows/auto-dependabot.yml
Comment (Romitou) [inline .github/workflows/auto-dependabot.yml:13]: What is the purpose of this step? If I'm not mistaken, it's only used to get some information from a Dependabot Pull Request, except that you never use this information through `metadata`.

File: .github/workflows/auto-dependabot.yml
Comment (Romitou) [inline .github/workflows/auto-dependabot.yml:19]: Correct me if I'm wrong, but the `success()` condition only applies to jobs where it is used. Wouldn't it be better to be sure that the compilation and unit tests are passed before merging on the master branch?

**Review by Romitou — COMMENTED** [review]
Hey! 👋

I'm not sure if the dependency auto-merge workflow would be a good idea. Firstly, I think it's good to always check the updates of dependencies, since probably some special cases may not be covered by unit testings, and also some dependencies could (although the risk is low) introduce malicious changes which would then be pushed directly to the master branch.

Here is my point of view, what do you think? 😄

File: .github/workflows/auto-dependabot.yml
Comment (TheLimeGlass) [reply .github/workflows/auto-dependabot.yml:13]: This is what gives Dependabot the lookup https://github.com/dependabot/fetch-metadata

File: .github/workflows/auto-dependabot.yml
Comment (TheLimeGlass) [reply .github/workflows/auto-dependabot.yml:19]: This is what the check does. If the tests fail, dependabot will make a pull request stating that updating this dependency caused an error.

**Comment by TheLimeGlass** [comment]
> Hey! 👋
> 
> I'm not sure if the dependency auto-merge workflow would be a good idea. Firstly, I think it's good to always check the updates of dependencies, since probably some special cases may not be covered by unit testings, and also some dependencies could (although the risk is low) introduce malicious changes which would then be pushed directly to the master branch.
> 
> Here is my point of view, what do you think? 😄

At the exact opposite, if a vulnerability like log4j was to come to fruition, dependabot would instantly update to the latest version that would have a patch for the vulnerability.

**Comment by TPGamesNL** [comment]
I agree that we should not automatically merge these. If there happens to be a critical vulnerability, we will need to take manual steps either way (i.e. release a new version), so it'd be safer to manually review these PRs.

**Review by TPGamesNL — APPROVED** [review]

**Comment by TheLimeGlass** [comment]
Waiting on reviews and pull https://github.com/SkriptLang/Skript/pull/4845 before merge.

**Review by APickledWalrus — APPROVED** [review]
Sounds good to me!

**Review by AyhamAl-Ali — APPROVED** [review]

**Pull request merged**
